src/vuln_analysis/server/webhooks/auth.py

```
@@ -121,13 +121,10 @@ async def _refresh(self) -> str:
 
         if response.status_code == 429:
-            # Honor Retry-After when present, then surface an error so the caller retries the
-            # whole delivery with its own backoff rather than hammering the token endpoint.
+            # Surface the retry hint without sleeping while the refresh lock is held. The
+            # delivery loop applies its own backoff after this exception releases the lock.
             retry_after = _parse_retry_after(response.headers.get("Retry-After"))
-            if retry_after:
-                logger.warning("Token endpoint rate-limited (429); backing off %.1fs", retry_after)
-                await asyncio.sleep(retry_after)
-            raise OAuth2Error("Token endpoint rate-limited (429)")
+            raise OAuth2Error(f"Token endpoint rate-limited (429); retry_after={retry_after:.1f}s")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)